src/hook/useFetchDappList.tsx

```diff
@@ -0,0 +1,42 @@
+import {useEffect, useState} from 'react';
+
+interface FetchResult<T> {
+  data: T | null;
+  loading: boolean;
+  error: Error | null;
+}
+
+function useFetchDappList<T>(
```

Comment:
Use `useQuery` hook to implement fetch with caching... (refer to `useSupportedTokens`)